src/Downloader.ts

```diff
@@ -184,45 +182,61 @@ class Downloader {
     return `${cachedPart}${path}`;
   }
 
-  public async checkCapabilities(): Promise<void> {
-    // check if RESTBase-powered API available
-    try {
-      const restApiMainPageQuery = await this.getJSON<any>(`${this.baseUrl}${encodeURIComponent(this.mw.metaData.mainPage)}`);
-      this.mwCapabilities.restApiAvailable = !!restApiMainPageQuery.lead;
-    } catch (err) {
-      this.mwCapabilities.restApiAvailable = false;
-      logger.warn(`Failed to get remote Rest API`);
+  public async setBaseUrls() {
```

Comment:
This function as still many problems:
* `isMobileApiAvailable` should be `isMobileRestApiAvailable`
* `isDesktopApiAvailable` should be `isDesktopRestApiAvailable`
* `isVeApiAvailable` is never used (but it should)
* Please do one logic bloc with the necessary `if` and `else` to set `this.baseUrl` ... and an other one to set `this.baseUrlForMainPage`. That way each logic is easly understandable and I don't have to read the whole function code.